src/lem/symbol.rs

```diff
@@ -1,15 +1,26 @@
-#[derive(Clone, PartialEq, Eq, Hash)]
+use super::AString;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub enum Symbol {
-    Sym(Vec<String>),
-    Key(Vec<String>),
+    Sym(Vec<AString>),
+    Key(Vec<AString>),
+}
+
+impl std::fmt::Display for Symbol {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Sym(vec) => write!(f, ".{}", vec.join(".")),
+            Self::Key(vec) => write!(f, ":{}", vec.join(".")),
+        }
+    }
```

Comment:
If this `Symbol` type is intended to be related to Lurk symbols, *please* use the existing type. Rolling your own 'similar' formatting for these 'similar to Symbols' types is just going to lead to gross or subtle bugs and/or waste time.

Reply:
The plan that I had was to use this file while the mega PR isn't merged, since the `Symbol` API there is pretty much the same as this one. In fact, I copy/pasted it from the mega PR

Reply:
What I see there is quite different, but maybe it's a snapshot. Regardless, let's make a plan to start agreeing with Lurk as soon as the mega-PR merges.